Packs/Yara/Scripts/YaraScan/YaraScan.py

```diff
@@ -42,16 +42,29 @@ def main():
                 "HasMatch": False,
                 "HasError": False,
                 "MatchCount": 0,
-                "Matches": list(),
-                "Errors": list()
+                "Matches": [],
+                "Errors": []
             }
             try:
-                cRule = yara.compile(source=yaraRuleRaw)
-            except Exception as err:
-                thisMatch['HasError'] = True
-                thisMatch['Errors'].append(str(err))
-                entries.append(thisMatch)
-                continue
+                rules = json.loads(yaraRuleRaw)
```

Comment:
Why would a rule file be in JSON format and not YARA format?

YARA:
```
rule rule_1 : tag{ ... }

rule rule_2{...}
```

This isn't a JSON file so it will always throw an exception. The original script supports multiple rules in a single file. It does not support multiple sources. We can add that support by adding a new argument named `filepaths` or better yet `sources`. These should be multiple entry ids as we are running through XSOAR.

Reply:
An entry ID is always a string, you need to open it within the script, see `line 20`. 

Either the script is called directly or via another script it has 2 inputs:
1. The rules.
  1a. A string of Rules (This is handled today via `yaraRule`)
  1b. A list of entry IDs which are yar files. (Not handled today)
2. Files to scan.

In the case of 1b the script should open the relevant files and compile them using the dictionary structure which should be built within the script.